File: experimentos/piloto/sc/Carga.cs
using Solucion;

namespace Operaciones;

public class Carga
{
    public static ResultadoOperacion Ejecutar(int dni, string clave, decimal saldo)
    {
        ResultadoOperacion ope = Soporte.ValidarAdministrador(dni, clave);
        if (!ope.OK) return ope;

        // Obtiene el cajero
        ResultadoOperacionLeer<Cajero> opeCajero = DominioRep.Leer<Cajero>(Tipos.Cajero, 1);
        if (!opeCajero.OK) return new ResultadoOperacion(opeCajero.CodigoError, opeCajero.Mensaje);
        Cajero cajero = opeCajero.Objeto;

        // Carga
        ope = cajero.IngresarDineroAlCajero(saldo);
        if (!ope.OK) return ope;

        // Persiste
        ope = DominioRep.Guardar(cajero);
        return ope;
    }

}

File: solucion_modelo/operaciones/carga/carga.py
from operaciones.carga.parametros_carga import ParametrosCarga
from tipos import RESULTADO
from estado import Estado
from utiles import es_administrador


def carga(solicitud: ParametrosCarga):
    if not es_administrador(solicitud.dni, solicitud.clave):
        return RESULTADO.UsuarioNoHabilitado

    estado = Estado()

    estado.saldo += solicitud.saldo

    estado.guardar()

    return RESULTADO.OK
